### ISSUE TITLE
Does rust-zmq work on Windows?

### ISSUE THREAD

**Issue by dkrikun** [erickt/rust-zmq#44]
Thanks :smile:

**Comment by amosbird** [comment]
I've add some posix88 symbols to make rust-zmq compile. And I'm using zeromq's windows dlls. Running the demo I get "Process didn't exit successfully: `target\debug\test-zmq.exe` (exit code: -1073741515)".

**Comment by pfernie** [comment]
I've put together 2 simple patches that gets things working for me on Windows, but was hoping for a little feedback before submitting a PR. First, my test case is a very small Req/Rep server, but seems to work fine; @amosbird, if you would be interested in testing the changes with, or pointing me to, the failing demo you mention I would appreciate it.

The [first patch](https://github.com/pfernie/rust-zmq/commit/a107ea0af08e5150ce8f82bc291419a99bfa02b9) is simply as @pblasucci describes; defining some posix88 constants to allow this to compile under windows, which I defined based on MSVC's errno.h. I haven't been able to extensively test all errors, but some of the more common errors (EACCES, EADDRINUSE, etc.) seem to behave as expected (correct error messages, etc.).

The [second patch](https://github.com/pfernie/rust-zmq/commit/b8fa62936bad9f337f78d6fa71a0aa6eb6c28f2a) is against zmq-sys. To get this to link properly for me in windows, I had to make the `#[link]` directive conditional on `cfg(windows)` in ffi.rs. The 0mq Windows binary distribution follows the [boost naming convention](http://zeromq.org/distro:microsoft-windows), so the libraries are named e.g. `libzmq-v90-mt-4_0_4.dll`, so a `#[link(name = "zmq")]` directive was causing link failures for `-lzmq`. Furthermore, I had to add a snippet to my `~/.cargo/config` in order to get rustc to link properly:
    
    [target.x86_64-pc-windows-gnu.zmq]
    rustc-link-lib = ["zmq-v90-mt-4_0_4"]
    rustc-link-search = ["c:/Program Files/ZeroMQ 4.0.4/bin"]

Obviously, `ffi.rs` is generated by `rust-bindgen`, so this probably isn't the right fix longterm... this works for me for the moment, but I imagine the proper fix is either supplying two versions of `ffi.rs` (`ffi-win.rs`, `ffi-non-win.rs`) and conditionally including the correct one, or doing a build-time generation of the bindings which can take platform into account. Or perhaps I am overcomplicating this; I spent enough time fighting the linker that I would love to hear a more straightforward solution.
